### PR TITLE
[RW-10362][risk=no] Test:  Users who used Absorb can take annual renewal training using Absorb

### PR DESCRIPTION
**Pull request by NehaBroad**
I am still thinking about integration/acceptance test. In the meantime i wanted to add this test which test:

1) User has already completed RT AND  CT training
2) User is re-enrolled to RT Training
3) User completes RT Training

---
**PR checklist**

- [ ] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [ ] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [ ] I have manually tested this change and my testing process is described above.
- [ ] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [ ] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [ ] None of the above apply to this change.

### CODE REVIEW

**Review by Peter-Lavigne — APPROVED** [review]
This is great, thanks for doing this!

I think the test should be split into at least two cases (see my comment on that), but I'll let you decide whether that's necessary.

I think an integration test or documented behavior that the Absorb API returns the expected value is needed to ensure this test will actually catch regressions.

None of my comments block merging if you'd prefer to address them later, if at all.

File: api/src/test/java/org/pmiops/workbench/compliancetraining/ComplianceTrainingServiceTest.java
Comment (NehaBroad) [reply api/src/test/java/org/pmiops/workbench/compliancetraining/ComplianceTrainingServiceTest.java:236]: I wanted to ensure even if RT is null we do not do anything to CT

**Pull request merged**
